### PR TITLE
Fix broken links

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**
Example Rest API call to fix broken links -

```
// Sample Rest API call
curl --location 'https://www-dev.greenpeace.org/usa/wp-json/update_broken_links/v1/data' \
--header 'Content-Type: application/json' \
--data '{
  "pid": [2640],
  "spid": [11582],
  "mime_type": "pdf"
}'
{"status":"success","logs":["2640 1"],"data":{"pid":[2640],"spid":[11582],"mime_type":"pdf"}}
```

### CODE REVIEW

File: wp_all_import_functions.php [hunk 1]
```diff
@@ -1,6 +1,103 @@
...
+        wp_update_post( $updated_post );
+    }
+
+    return [
+        'postid' => $postid ,
+        'update_status' => $update_status,
+    ];
+}
+
+//add_action('pmxi_saved_post','post_saved',5,2);
```
Comment (sagarsdeshmukh) [inline wp_all_import_functions.php:100]: I comment this action hook as its more relevant and needed for fresh data import.

File: wp_all_import_functions.php [hunk 2]
```diff
@@ -1,6 +1,103 @@
...
+            continue;
+        }
+
+        $basename = basename( $media_file );
+        $basename = str_replace(' ', '-', $basename);
+        $basename = str_replace('%20', '-', $basename);
+        foreach ( $attachments as $attachment ) {
+            $guid = $attachment->guid;
+
+            /*// If same image/pdf imported twice, the stateless append -1 in file name.
```
Comment (sagarsdeshmukh) [inline wp_all_import_functions.php:72]: At this point its not needed but useful in future/other usecases

File: wp_all_import_functions.php [hunk 3]
```diff
@@ -1,6 +1,103 @@
...
+        $single_quote_media = array_pop( $single_quote_match_img );
+        $media_files = array_merge($media_files, $single_quote_media);
+    }
+
+    $update_status = 0;
+
+    // Fix broken url in Post content.
+    foreach ( $media_files as $media_file ) {
+        // If media file belongs to P4 stateless bucket, skip it.
+        if (strpos($media_file,"planet4-usa-stateless")) {
```
Comment (sagarsdeshmukh) [inline wp_all_import_functions.php:62]: If we rerun the rest API with the same param, this condition will skip already updated media.

File: wp_all_import_functions.php [hunk 4]
```diff
@@ -1,6 +1,103 @@
...
+    $content_post = get_post( $postid );
+    $content = $content_post->post_content;
+    $content = apply_filters( 'the_content', $content );
+
+    if ($mime_type === 'pdf') {
+        $attachments = get_attached_media( 'application/pdf', $spostid );
+        preg_match_all( '@href="([^"]+\.pdf|PDF)"@' , $content, $match_pdf );
+        $media_files = array_pop( $match_pdf );
+
+        // check for single quote pdf files.
```
Comment (sagarsdeshmukh) [inline wp_all_import_functions.php:41]: In some posts, the pdf/image link is enclosed in single quotes, the below check will take care such usecases.

**Review by comzeradd — APPROVED** [review]
👍🏻

**Pull request merged**